test/change-env-cwd.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+var Liftoff = require('../');
+var expect = require('chai').expect;
+var fs = require('fs');
+var path = require('path');
+var fixturesDir = path.resolve(__dirname, 'fixtures/change-env-cwd');
+
+describe('Change env.cwd', function() {
+
+  it('should set env.cwd to initial cwd with no option and no config flag', function(done) {
+    process.chdir(path.join(fixturesDir, 'a'));
+    var app = new Liftoff({ name: 'app' });
+    app.prepare({}, function(env) {
+      app.execute(env, function(env) {
+        expect(env.cwd).to.equal(path.join(fixturesDir, 'a'));
+        done();
+      });
+    });
+  });
+
+  it('should change env.cwd by a prepare option', function(done) {
+    process.chdir(path.join(fixturesDir, 'a'));
+    var app = new Liftoff({ name: 'app' });
+    app.prepare({ cwd: '../b' }, function(env) {
+      app.execute(env, function(env) {
+        expect(env.cwd).to.equal(path.join(fixturesDir, 'b'));
+        done();
+      });
+    });
+  });
+
+  it('should change env.cwd by a config file', function(done) {
+    process.chdir(path.join(fixturesDir, 'a'));
+    var app = new Liftoff({
+      name: 'app',
+      configFiles: {
+        '.app': { cwd: { path: '.' } },
+      },
+    });
+    app.prepare({}, function(env) {
+      var fp = env.configFiles['.app'].cwd;
+      var cfg = JSON.parse(fs.readFileSync(fp, 'utf8'));
+      env.cwd = cfg.cwd;
+      app.execute(env, function(env) {
+        expect(env.cwd).to.equal('../b');
+        done();
+      });
+    });
+  });
+
+  it('should change env.cwd by [1] a prepare option and [2] a config file in the directory moved by [1]', function(done) {
+    process.chdir(path.join(fixturesDir, 'a'));
+    var app = new Liftoff({
+      name: 'app',
+      configFiles: {
+        '.app': { cwd: { path: '.' } },
+      },
+    });
+    app.prepare({ cwd: '../b' }, function(env) {
+      var fp = env.configFiles['.app'].cwd;
+      var cfg = JSON.parse(fs.readFileSync(fp, 'utf8'));
+      env.cwd = cfg.cwd;
+      app.execute(env, function(env) {
+        expect(env.cwd).to.equal('./c');
+        done();
+      });
+    });
+  });
+
+  it('should change env.cwd by [1] a config file in initial cwd and [2] a config file in the directory moved by [1]', function(done) {
+    process.chdir(path.join(fixturesDir, 'a'));
+    var app = new Liftoff({
+      name: 'app',
+      configFiles: {
+        '.app': { cwd: { path: '.' } },
+      },
+    });
+
+    var cfgfiles0 = app.findConfigFiles({
```

Comment:
I'd like to avoid this having to be called outside of/before compare to achieve the results we desire.  I'm going to outline my thoughts in the top-level review.